airflow/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -206,15 +206,16 @@ def _resolve_connection(self) -> dict[str, Any]:
             spark_binary = self._spark_binary or extra.get("spark-binary", "spark-submit")
             if spark_binary not in ALLOWED_SPARK_BINARIES:
                 raise RuntimeError(
-                    f"The `spark-binary` extra can be on of {ALLOWED_SPARK_BINARIES} and it"
+                    f"The `spark-binary` extra can be one of {ALLOWED_SPARK_BINARIES} and it"
                     f" was `{spark_binary}`. Please make sure your spark binary is one of the"
                     " allowed ones and that it is available on the PATH"
                 )
             conn_spark_home = extra.get("spark-home")
             if conn_spark_home:
                 raise RuntimeError(
-                    "The `spark-home` extra is not allowed any more. Please make sure your `spark-submit` or"
-                    " `spark2-submit` are available on the PATH."
+                    "The `spark-home` extra is not allowed any more. Please make sure one of"
+                    f" {ALLOWED_SPARK_BINARIES} is available on the PATH, and set `spark-binary`"
```

Comment:
```suggestion
                    f" {ALLOWED_SPARK_BINARIES} is available on the PATH, or set `spark-binary`"
```

I think this is more… English?

Reply:
" if needed" on the end.  I don't think this code is looking for any of `spark_binary` on PATH, just one of them.  So, if your Spark distro has anything other than `spark-submit` as the executable, you'll need to set `spark_binary` and also ensure that the binary is on the PATH.

Reply:
OK